pkg/parser/buildah/parser.go

```diff
@@ -0,0 +1,206 @@
+type Info struct {
+	IgnoreLines      []int
+	From             map[string][]Command
+	FromValues       []FromValue
+	IgnoreBlockLines []int
+}
+
+const (
+	buildah = "buildah"
+)
+
+// Resolve - replace or modifies in-memory content before parsing
+func (p *Parser) Resolve(fileContent []byte, filename string) (*[]byte, error) {
+	return &fileContent, nil
+}
+
+// Parse - parses Buildah file to Json
+func (p *Parser) Parse(_ string, fileContent []byte) ([]model.Document, []int, error) {
+	var info Info
+	info.From = map[string][]Command{}
+
+	reader := bytes.NewReader(fileContent)
+	f, err := syntax.NewParser(syntax.KeepComments(true)).Parse(reader, "")
+
+	if err != nil {
+		return nil, []int{}, err
+	}
+
+	syntax.Walk(f, func(node syntax.Node) bool {
+		switch x := node.(type) {
+		case *syntax.Stmt:
+			info.getStmt(x)
+		case *syntax.Comment:
+			info.getIgnoreLines(x)
+		}
+		return true
+	})
+
+	// get kics-scan ignore-block related to from
+	info.ignoreFromBlock()
+
+	var documents []model.Document
+	var resource Resource
+	resource.CommandList = info.From
+	doc := &model.Document{}
+	j, err := json.Marshal(resource)
+	if err != nil {
+		return nil, []int{}, errors.Wrap(err, "failed to Marshal Buildah")
+	}
+
+	err = json.Unmarshal(j, &doc)
+	if err != nil {
+		return nil, []int{}, errors.Wrap(err, "failed to Unmarshal Buildah")
+	}
+
+	documents = append(documents, *doc)
+
+	sort.Ints(info.IgnoreLines)
+
+	return documents, info.IgnoreLines, nil
+}
+
+func (i *Info) getStmt(stmt *syntax.Stmt) {
+	if cmd, ok := stmt.Cmd.(*syntax.CallExpr); ok {
+		args := cmd.Args
+
+		// get kics-scan ignore-block related to command + get command
+		stCommand := i.getStmtInfo(stmt, args)
+
+		if stCommand.Cmd == "buildah from" {
+			fromValue := FromValue{
+				Value: stCommand.Value,
+				Line:  stCommand.StartLine,
+			}
+			i.FromValues = append(i.FromValues, fromValue)
+		}
+
+		if stCommand.Cmd != "" {
+			if len(i.FromValues) != 0 {
+				v := i.FromValues[len(i.FromValues)-1].Value
+				i.From[v] = append(i.From[v], stCommand)
+			}
+		}
+	}
+}
+
+func (i *Info) getStmtInfo(stmt *syntax.Stmt, args []*syntax.Word) Command {
+	var command Command
+	minimumArgs := 2
+
+	if len(args) > minimumArgs {
+		if getWordValue(args[0]) == buildah {
+			cmd := "buildah " + strings.TrimSpace(getWordValue(args[1]))
+			fullCmd := strings.TrimSpace(getFullCommand(args))
+			value := strings.TrimPrefix(fullCmd, cmd)
+			start := int(args[0].Pos().Line())
+			end := int(args[len(args)-1].End().Line())
+
+			command = Command{
+				Cmd:       cmd,
+				Original:  fullCmd,
+				StartLine: start,
+				EndLine:   end,
+				Value:     strings.TrimSpace(value),
+			}
+
+			// get kics-scan ignore-block comments
+			i.getIgnoreBlockLines(stmt.Comments, start, end)
+
+			return command
+		}
+	}
+	return command
+}
+
+func getWordValue(wd *syntax.Word) string {
+	printer := syntax.NewPrinter()
+	var buf bytes.Buffer
+
+	printer.Print(&buf, wd)
```

Comment:
We should handle this error also, if we are only using the printer to place the wd inside a buffer can't we use the `fmt` package with the `Frintf()` method?

Same for line 171